### DISCUSSION TITLE
Are frames dropping at all?

### DISCUSSION THREAD

**Discussion by mkrn** [HaishinKit/HaishinKit.swift#1252] [Q&A]
Curious if right now the framework is dropping frames if connection is no good and bitrate is too high.
Had an issue with adaptive bitrate implementation and noticed that framework is sending old frames, putting transcoding very much behind.

noticed this delegate method available:
```
func streamWillDropFrame(_ stream: NetStream) -> Bool {
        // Implementation for determining whether to drop a video frame
        return false
    }
```
Is this method to decide to drop a frame from being sent out?
But what can you use in `stream` to decide if you need to drop a frame?
My guess is in RTMPConnection one could at least use `currentBytesOutPerSecond` to decide if they should drop a frame

Any other methods to control how big the buffer is and to force the framework skip frames?

**Comment by shogo4405** [comment]
Yes, it serves as a delegate to change the frame rate before encoding. 
If streamWillDropFrame always returns true, it will stop the transmission of all frames.

**Reply by shogo4405** [reply]
You might find this article helpful for bitrate adjustment.
https://gcore.com/blog/how-to-create-a-mobile-streaming-app-on-ios/
